src/Features/Core/Portable/Common/NavigationOperation.cs

```diff
@@ -34,7 +35,10 @@ public override void Apply(Workspace workspace, CancellationToken cancellationTo
             if (workspace.CanOpenDocuments)
             {
                 var navigationService = workspace.Services.GetService<IDocumentNavigationService>();
-                navigationService.TryNavigateToPosition(workspace, _documentId, _position, cancellationToken);
+                var threadingService = workspace.Services.GetService<IWorkspaceThreadingServiceProvider>();
```

Comment:
this is a new concept @jasonmalinowski . not sure if there's any better way here.

Reply:
Yeah, I don't have many better options other than leave the sync method on the IDocumentNavigationService and then inline this, which of course kinda undoes what you're trying to do. So unless @sharwell has a better idea, I'm OK with this in the "I will pinch my nose and sign off because I don't have a better idea".